components/bsx/Offer/MasterOfferTable.vue

```diff
@@ -0,0 +1,201 @@
+import PrefixMixin from '~/utils/mixins/prefixMixin'
+import SubscribeMixin from '~/utils/mixins/subscribeMixin'
+import offerListUser from '@/queries/subsquid/bsx/offerListUser.graphql'
+import { notificationTypes, showNotification } from '~/utils/notification'
+import { isAddress } from '@polkadot/util-crypto'
+import ChainMixin from '~/utils/mixins/chainMixin'
+import { SelectedOfferType } from '~/utils/offerStatus'
+
+const components = {
+  Loader: () => import('@/components/shared/Loader.vue'),
+  CollapseCardWrapper: () =>
+    import('@/components/shared/collapse/CollapseCardWrapper.vue'),
+  StatsOverview: () => import('~/components/bsx/Offer/StatsOverview.vue'),
+  AddressInput: () => import('@/components/shared/AddressInput.vue'),
+  OffersUserTable: () => import('@/components/bsx/Offer/OffersUserTable.vue'),
+  MyOffer: () => import('@/components/bsx/Offer/MyOffer.vue'),
+  OfferTable: () => import('@/components/bsx/Offer/OfferTable.vue'),
+}
+
+@Component({ components })
+export default class MasterOfferTable extends mixins(
+  AuthMixin,
+  MetaTransactionMixin,
+  PrefixMixin,
+  SubscribeMixin,
+  ChainMixin
+) {
+  protected createdOffers: Offer[] = []
+  protected total = 0
+  protected destinationAddress = ''
+  protected accountIdChanged = ''
+  protected incomingOffers: Offer[] = []
+  protected SelectedOfferType = SelectedOfferType
+  protected skipUserOffer = false // skip fetching with id when this variable is true
+
+  public async created() {
+    this.checkQueryParams()
+  }
+
+  fetch() {
+    this.fetchCreatedOffers()
+  }
+
+  get isOfferDropdownDisabled(): boolean {
+    return this.skipUserOffer
+  }
+
+  get ss58Format(): number {
+    return this.chainProperties?.ss58Format
+  }
+
+  get selectedOfferType(): string {
+    return (this.$route.query.tab as string) || SelectedOfferType.ALL
+  }
+
+  set selectedOfferType(val) {
+    const { target } = this.$route.query
+    this.$router.replace({
+      query: { target, tab: val },
+    })
+  }
+
+  public offersIncomingUpdate(data) {
+    this.incomingOffers = data.offers
+  }
+
+  public getOfferTypeOptions() {
+    return [
+      {
+        type: SelectedOfferType.ALL,
+        label: 'All Offers',
+      },
+      {
+        type: SelectedOfferType.CREATED,
+        label: 'Offers Created',
+      },
+      {
+        type: SelectedOfferType.INCOMING,
+        label: 'Offers Incoming',
+      },
+    ]
+  }
+
+  protected checkQueryParams() {
+    const { query } = this.$route
+    if (query.target) {
+      const hasAddress = isAddress(query.target as string)
+      if (hasAddress) {
+        this.destinationAddress = query.target as string
+        this.accountIdChanged = query.target as string
+      } else {
+        showNotification('Unable to use target address', notificationTypes.warn)
+      }
+    } else {
+      this.skipUserOffer = true
+    }
+  }
+
+  public checkOfferForAddress(skipAddress?) {
+    this.skipUserOffer = skipAddress
```

Comment:
the method is called once without giving any value to `skipAddress`so I'm not sure what will happen when setting `skipUserOffer` to optional value. I think its better to just set it to false as default value
```suggestion
  public checkOfferForAddress(skipAddress = false) {
    this.skipUserOffer = skipAddress
```